### PR TITLE
Feature/shutdown dialog

### PR DESCRIPTION
**Pull request by theCalcaholic**
When pressing the power button in ncp-web, show a menu with options 'shut down' and 'reboot' (compare #125).

### CODE REVIEW

**Comment by nachoparker** [comment]
great! I'll look at it soon

**Comment by theCalcaholic** [comment]
Could you test it once (just see if shut down and reboot work as intended)? I don't completely trust my docker environment and have no NCP device at hand atm.

**Comment by nachoparker** [comment]
sure, I test everything before merging it

File: nextcloudpi.sh [hunk 1]
```diff
@@ -120,7 +120,7 @@ sleep 0.5 && echo "" > /run/ncp.log
 exit $RET
 EOF
   chmod 700 /home/www/ncp-launcher.sh
-  echo "www-data ALL = NOPASSWD: /home/www/ncp-launcher.sh , /sbin/halt" >> /etc/sudoers
+  echo "www-data ALL = NOPASSWD: /home/www/ncp-launcher.sh , /sbin/halt, /sbin/reboot" >> /etc/sudoers
```
Comment (nachoparker) [inline nextcloudpi.sh]: this needs to be included too for old images, so it should be duplicated in `update.sh`

File: ncp-web/index.php [hunk 2]
```diff
@@ -169,6 +169,13 @@
 
   </div>
 
+  <div id="poweroff-dialog" class='dialog primary' style='display:none;'>
```
[the remark this answers is not in the document]
Comment (theCalcaholic) [reply ncp-web/index.php]: That's how the minifiedjs/jquery methods do it. If we do it like this it's compatible with `$(element).hide()` and 
`.show()`.

File: nextcloudpi.sh (hunk 1 above)
Comment (theCalcaholic) [reply nextcloudpi.sh]: Ah right. I thought I had already added this. :+1:

Comment (nachoparker) [reply nextcloudpi.sh]: you added it in nextcloudpi.sh which is only used for building new images. If we are to update currently running images with the new feature through `ncp-update`, then we need to do this in `update.sh`

other changes, like any changes to files in `bin`, `ncp-web` and `etc` are overwritten automatically without more intervention, but this is `/etc/sudoers` so we have to do it explicitly.

File: ncp-web/index.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (nachoparker) [reply ncp-web/index.php]: just use the `hidden class` and get rid of the inline style ;)

[the remark this answers is not in the document]
Comment (theCalcaholic) [reply ncp-web/index.php]: And that works with said functions? Oo
Alright then.

[the remark this answers is not in the document]
Comment (nachoparker) [reply ncp-web/index.php]: like here
https://github.com/nextcloud/nextcloudpi/blob/master/ncp-web/index.php#L157

File: nextcloudpi.sh (hunk 1 above)
Comment (theCalcaholic) [reply nextcloudpi.sh]: Yes, I thought I had added it twice. ;)
But apparently not so.

Comment (nachoparker) [reply nextcloudpi.sh]: :D looking great, once those two little things are fixed, I will build a devel image with `build_devel.sh` and test it.

I want to make a command to update from devel, so we don't need to use QEMU and generate images to test.

**Comment by nachoparker** [comment]
pushing it to devel for the final testing

thank you!

**Pull request closed without merging**